app/models/medication.rb

```diff
@@ -37,4 +39,8 @@ class Medication < ActiveRecord::Base
   def active_reminders
     [refill_reminder, take_medication_reminder].select(&:active?)
   end
+
+  def should_generate_new_friendly_id?
```

Comment:
you could put all of these `should_generate_new_friendly_id?` etc into a  FriendlyId concern if you want: http://stackoverflow.com/questions/14541823/how-to-use-concerns-in-rails-4